blivet/__init__.py

```diff
@@ -873,7 +873,11 @@ def clearPartitions(self):
             if should_clear:
                 self.recursiveRemove(disk)
 
-            if zerombr or should_clear:
+            if disk.readonly:
+                raise ProtectedDiskError("cannot clear '%s': disk is read only" % disk.name)
```

Comment:
We shouldn't raise an exception here in the general case. Consider the following kickstart:

```zerombr
clearpart --initlabel --all
autopart
```

If there are several disks and only one of them is read-only, we should just not use the read-only disk. We already do this for `clearpart`, but not `zerombr`. I think the most you should do in these cases is log a warning and continue without initializing the disk. Also, I'm pretty sure that `disk.readonly` implies `disk.protected`, so you wouldn't need to check both of them.

Reply:
Fixed. I have checked that KS configuration. It behaves like you described. Apparently #636 does not apply for this situation. I have replaced the exception with a warning. Also removed `disk.readonly` and updated the tests accordingly.